turbine/src/quic_endpoint.rs

```diff
@@ -18,20 +18,20 @@ use {
         collections::{hash_map::Entry, HashMap},
         io::Error as IoError,
         net::{IpAddr, SocketAddr, UdpSocket},
-        ops::Deref,
         sync::Arc,
     },
     thiserror::Error,
     tokio::{
         sync::{
-            mpsc::{Receiver as AsyncReceiver, Sender as AsyncSender},
-            RwLock,
+            mpsc::{error::TrySendError, Receiver as AsyncReceiver, Sender as AsyncSender},
+            Mutex, RwLock as AsyncRwLock,
         },
         task::JoinHandle,
     },
 };
 
-const CLIENT_CHANNEL_CAPACITY: usize = 1 << 20;
+const CLIENT_CHANNEL_BUFFER: usize = 1 << 14;
+const ROUTER_CHANNEL_BUFFER: usize = 64;
```

Comment:
Remind me, would we reasonably expect to hit "capacity" on these channels? I see that you handle the errors for `TrySendError::Full` with an error, but wondering how tight these are?

Also, would be potentially be more susceptible to these backing up on smaller clusters (ie some of the <= 10 node clusters we have)? Maybe not a concern as QUIC-turbine is still a work-in-progress, but if we end up going this way I could also see these constants not getting updated until something breaks.

Reply:
I wouldn't expect to hit `CLIENT_CHANNEL_BUFFER` (as long as tokio scheduling works decently) because that immediately forwards packets to the router channels.
`ROUTER_CHANNEL_BUFFER` depends on how long it takes to establish a connection with a remote node and how many shreds should be forwarded to that node in the meanwhile. So it depends on peer-to-peer latency and stakes.
With both we will drop shreds (i.e. load shedding) when the buffer capacity is hit (broadcast stage will do back-pressure). So I believe it is fine to hit the buffer capacity anyways.

Following commits will add some metrics, and then testing on testnet we can pick some better values for these. Ideally these should be the smallest feasible values.